### ISSUE TITLE
Limit all headings to 40rem, same as text

### ISSUE THREAD

**Issue by lyubomir-popov** [canonical/vanilla-framework#5200]
We've received complaints that headings take up more horizontal width than text in the documentation layout, where text is not constrained to narrow columns, but spans the full available width until it hits the 40em max-width. For bigger font-size headings, that's a lot wider than surronding paragraphs. I propose we limit everything to 40rem as opposed to 40em, so things look tidier.

**Comment by bartaz** [comment]
You know it means all the headings everywhere will be restricted? Is that what we want?

<img width="1312" alt="image" src="https://github.com/canonical/vanilla-framework/assets/83575/3f755c7b-158f-4097-a5ae-4e1a295c1f33">

**Comment by bartaz** [comment]
@lyubomir-popov is this something we still want to do (knowing that it may affect many existing headings in unexpected ways)?

**Comment by lyubomir-popov** [comment]
good point, hadn't considered that. What would you recommend for the docs problem specifically? Can we have maybe recommend using the utility (.p-max-width IIRC) on the parent?

**Comment by bartaz** [comment]
> good point, hadn't considered that. What would you recommend for the docs problem specifically? Can we have maybe recommend using the utility (.p-max-width IIRC) on the parent?

May be a good call. We already use it to limit width of lists, so we can include headings into that.

I'm triaging it to be fixed this way then.
